substrate-archive/src/archive.rs

```diff
@@ -32,129 +32,146 @@ use sp_runtime::{
 use substrate_archive_backend::{runtime_api, ReadOnlyBackend, TArchiveClient};
 use substrate_archive_common::{util, ReadOnlyDB, Result};
 
-use crate::{actors::System, traits};
+use crate::{
+	actors::{ActorContext, System},
+	migrations,
+};
 
 const CHAIN_DATA_VAR: &str = "CHAIN_DATA_DB";
 const POSTGRES_VAR: &str = "DATABASE_URL";
 
 /// The recommended open file descriptor limit to be configured for the process.
 const RECOMMENDED_OPEN_FILE_DESCRIPTOR_LIMIT: u64 = 10_000;
 
-pub struct Builder<B, R, D, DB> {
+#[async_trait::async_trait(?Send)]
+pub trait Archive<B: BlockT + Unpin, D: ReadOnlyDB>
```

Comment:
Some docs would be good. I know many public items are not documented but as we move code around we should try to improve this.

Reply:
okay, I will add some docs later